algo/src/main/java/com/hartwig/serve/extraction/gene/GeneLevelExtractor.java

```diff
@@ -106,6 +114,34 @@ static boolean geneInDriverGenes(@NotNull List<DriverGene> driverGenes, @NotNull
         return false;
     }
 
+    @Nullable
+    @VisibleForTesting
+    GeneAnnotation extractProteinEvent(@NotNull String gene, @NotNull EventType type) {
+        boolean geneInDriverGenesDatabase = geneInDriverGenes(driverGenes, gene);
+        GeneEvent geneEvent;
+        if (type == EventType.ABSENCE_OF_PROTEIN) {
+            geneEvent = GeneEvent.ABSENCE_OF_PROTEIN;
+        } else if (type == EventType.PRESENCE_OF_PROTEIN) {
+            geneEvent = GeneEvent.PRESENCE_OF_PROTEIN;
+        } else {
+            throw new IllegalStateException("Invalid event type passed into protein event extractor function: " + type);
+        }
+
+        if (!geneInDriverGenesDatabase && driverInconsistencyMode.isActive()) {
+            if (driverInconsistencyMode == DriverInconsistencyMode.WARN_ONLY) {
+                LOGGER.warn("{} event {} on {} is not included in driver catalog and won't ever be reported.", geneEvent, type, gene);
+            } else if (driverInconsistencyMode == DriverInconsistencyMode.FILTER) {
+                LOGGER.info("{}  event filtered -- {} on {} is not included in driver catalog and won't ever be reported.",
+                        geneEvent,
+                        type,
+                        gene);
+                return null;
+            }
+        }
+
+        return ImmutableGeneAnnotationImpl.builder().gene(gene).event(geneEvent).build();
+    }
+
```

Comment:
Have rewritten part of this, but don't see a unit test for this function yet (even though it is tagged as 'VisibleForTesting')!

Reply:
I have indeed forgotten to test this function. I have implemented a test for this!